internal/agent/executor/cache.go

```diff
@@ -114,6 +115,15 @@ func (executor *Executor) DownloadCache(
 
 	cachePopulated, cacheAvailable := executor.tryToDownloadAndPopulateCache(ctx, logUploader, commandName, cacheHost, cacheKey, baseFolder)
 
+	if !cacheAvailable && instruction.OptimisticallyRestoreOnMiss {
+		logUploader.Write([]byte("\nWasn't able to find the exact cache! Requesting the last available one..."))
+		cacheInfo := executor.findLatestAvailableCache(ctx, logUploader, commandName)
+		if cacheInfo != nil {
```

Comment:
It would probably be a good idea to handle `cacheInfo == nil` to and print a log entry explaining what happened.